### ISSUE TITLE
Efficiently get the check results of a small list of packages

### ISSUE THREAD

**Issue by wlandau** [r-universe-org/help#370]
For both https://github.com/r-releases/help/issues/6 and https://github.com/r-releases/help/issues/10, it would help to know the most efficient way of scraping check results from a small list of packages. Given a character vector of package names, I would like to be able to efficiently get a bottom-line pass/fail verdict for each one. Is there somewhere in https://r-universe.dev/stats that we could scrape? Or would it be better to scrape https://github.com/r-universe/r-releases/actions?

**Comment by jeroen** [comment]
I could add an API to do provide the information you need without scraping. However it is not clear what you mean by "the check result", because we run on many platforms, multiple versions of R. Builds can fail for many reasons, some may be intermittent, or only one one platform.

**Comment by wlandau** [comment]
> I could add an API to do provide the information you need without scraping.

That would be amazing!

> However it is not clear what you mean by "the check result"

For https://github.com/r-releases/help/issues/6, I think it would be enough to get the overall pass/fail outcome for all platforms within a single user-specified (default: current) version of R.

**Comment by wlandau** [comment]
> single user-specified version of R (wrappers in r-releases could be responsible for setting getRversion() as a default).

Hmmm... for https://github.com/r-releases/help/issues/6#issuecomment-1975118710, maybe the capability to scrape multiple versions of R in the same API call might help, as well as scrape a medium-sized list of packages with few (paginated) API calls.

**Comment by wlandau** [comment]
For a first attempt at production, we are moving away from the “install_safe()” idea (https://github.com/r-releases/help/issues/6) in favor of the dual-repo option ( https://github.com/r-releases/help/issues/10). For the API you proposed, this could potentially make things a lot simpler.

And for either option, I think we only need information about specific failing checks, omitting mentions of successful checks and successful packages. This could dramatically reduce the size of the payload and allow the API to report more detailed information. As long as we can give users a full list of checks and tell them how often they run, this will come together nicely.

**Comment by wlandau** [comment]
> I think we only need information about specific failing checks, omitting mentions of successful checks and successful packages. 

On reflection, we also need confirmation that the checks actually ran for the version currently in QA. This should prevent a new package or new version from moving to production until we know the results of the latest checks. Depending on the order of operations, we might already have enough information without the proposed API in this thread. @jeroen, if a package shows up in https://r-multiverse.r-universe.dev/src/contrib/PACKAGES?fields=RemoteSha, can we assume the check results from https://r-multiverse.r-universe.dev/builds are already up to date with the specific `RemoteSha` listed in `contrib/PACKAGES`?

**Comment by wlandau** [comment]
Related: in R-multiverse, it would be best if the production universe has explicit Git references rather than `"branch": "*release"`. In the latter case, a new and possibly broken package version could go to production without any automated gatekeeping. Can the `"branch"` field understand `RemoteSha`s?

**Comment by jeroen** [comment]
Yes `branch` may also be a sha, it can be any git ref. In retrospect `branch` should have been called `ref`; initially we only supported branches but now you can do any ref.

**Comment by jeroen** [comment]
> if a package shows up in https://r-multiverse.r-universe.dev/src/contrib/PACKAGES?fields=RemoteSha, can we assume the check results from https://r-multiverse.r-universe.dev/builds are already up to date with the specific RemoteSha listed in contrib/PACKAGES?

Yes in r-universe everything is deployed at once. So the checks and binaries are always in sync with the source package.

**Comment by wlandau** [comment]
That's fantastic! In that case, this new package check API can definitely just report specific failures. R-multiverse infrastructure can then match the package names from the API against the names from `contrib/PACKAGES` and trust the `RemoteSha`s etc.
